### PR TITLE
track task run params in state details to avoid quantum entanglement

### PR DESCRIPTION
**Pull request by abrookins**
…with results

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by abrookins** [comment]
I kept breaking things until a task ran, check it out:

```
fastapi-user-signups-api-1        | 05:54:26.282 | INFO    | prefect.engine - Submitted run of task send_confirmation_email with ID 363b457d-8a4f-4925-a614-ee4eadffa84d
fastapi-user-signups-tasks-1      | 05:54:30.504 | INFO    | prefect.task_server - Scheduled(message=None, type=SCHEDULED, result=None)
fastapi-user-signups-tasks-1      | 05:54:31.337 | INFO    | Flow run '<unknown>' - Created task run 'send_confirmation_email-b4aa425c' for task 'send_confirmation_email'
fastapi-user-signups-tasks-1      | 05:54:31.338 | INFO    | Flow run '<unknown>' - Executing 'send_confirmation_email-b4aa425c' immediately...
fastapi-user-signups-mailboi-1    | time="2024-01-26T05:54:31Z" level=error msg="Handled request" bytes-in=116 bytes-out=635 end="2024-01-26T05:54:31Z" headers="map[Accept:[*/*] Accept-Encoding:[gzip, deflate] Connection:[keep-alive] Content-Length:[116] Content-Type:[application/json] User-Agent:[python-httpx/0.26.0]]" host=mailboi latency=3.97225ms method=POST path=/send-mail remote-ip=172.24.0.6 start="2024-01-26T05:54:31Z" status=666 uri=/send-mail
fastapi-user-signups-tasks-1      | 05:54:31.444 | INFO    | Task run 'send_confirmation_email-b4aa425c' - Finished in state Completed()
```

File: src/prefect/task_server.py [hunk 1]
```diff
@@ -125,14 +124,25 @@ async def _submit_pending_task_runs(self, task_run_response):
...
             parameters = {}
-            if isinstance(task_run.state.data, LiteralResult):
-                parameter_id = UUID(task_run.state.data.value)
+            if hasattr(task_run.state.state_details, "task_parameters_id"):
+                parameters_id = task_run.state.state_details.task_parameters_id
                 task.persist_result = True
                 factory = await ResultFactory.from_task(task)
-                parameters = await factory.read_parameters(parameter_id)
+                try:
+                    parameters = await factory.read_parameters(parameters_id)
```
Comment (abrookins) [inline src/prefect/task_server.py:135]: We probably don't want this exception handler, I was just trying to cut down on the error output while the task engine couldn't find the files in storage.

File: src/prefect/client/schemas/objects.py [hunk 2]
```diff
@@ -113,6 +113,7 @@ class StateDetails(PrefectBaseModel):
     refresh_cache: bool = None
     retriable: bool = None
     transition_id: Optional[UUID] = None
+    task_parameters_id: Optional[UUID] = None
```
Comment (chrisguidry) [inline src/prefect/client/schemas/objects.py:116]: Ha, this is exactly what I was planning to do this morning!

File: src/prefect/results.py [hunk 3]
```diff
@@ -392,6 +392,7 @@ async def store_parameters(self, identifier: UUID, parameters: Dict[str, Any]):
         ), "Unexpected storage block ID. Was it persisted?"
         data = self.serializer.dumps(parameters)
         blob = PersistedResultBlob(serializer=self.serializer, data=data)
+        logger.info(f"STORING PARAMS {self.storage_block} {blob}")
```
Comment (chrisguidry) [inline src/prefect/results.py:395]: We can probably drop this

File: src/prefect/server/database/orm_models.py [hunk 4]
```diff
@@ -253,6 +253,7 @@ def data(self):
         if not self.result_artifact_id:
             # do not try to load the relationship if there's no artifact id
             return None
+        print("HEY ", self.id)
```
Comment (chrisguidry) [inline src/prefect/server/database/orm_models.py:256]: Hey yourself

**Review by chrisguidry — APPROVED** [review]
This looks great, it was exactly what I was planning to do today, to add a new attribute for ourselves on `state_details`.  I was running short on time yesterday afternoon to explain, but the whole `data` thing was totally a hack just to get something stored server-side, but this is definitely the better way

**Comment by chrisguidry** [comment]
I ran through this locally with the task-scheduling-examples repo and it works well!

**Pull request merged**